var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
By the way, in my new `PythonPackage` class, you'll notice quite a bit of repeated code from [this line](https://github.com/LLNL/spack/pull/2709/files#diff-37f9ae38dd9352ddf85787d094aaf273R114) to the end of the file. If the user wants to use a phase not in this list, they need to repeat the same code again in their `package.py`. Every phase of `PythonPackage` looks the same:
```python
def <phase>(self, spec, prefix):
    args = self.<phase>_args(spec, prefix)
   
    self.setup_py('<phase>', *args)

def <phase>_args(self, spec, prefix:
    return []
```
Basically, if a user wants:
```python
phases = ['configure', 'build', 'install']
```
I want Spack to run:
```bash
python setup.py configure
python setup.py build
python setup.py install
```
But I also want them to have access to `<phase>_args` in case they need to add arguments. Do either of you know any smart way to do this aside from defining all of the functions in `PythonPackage`?